src/lexers/jsx-lexer.js

```diff
@@ -1,120 +1,69 @@
-import * as acorn from 'acorn-jsx'
+import * as walk from 'acorn/dist/walk'
 import assert from 'assert'
-import HTMLLexer from './html-lexer'
-import BaseLexer from './base-lexer'
 import { ParsingError } from '../helpers'
 
-export default class JsxLexer extends HTMLLexer {
-  constructor(options = {}) {
-    options.attr = options.attr || 'i18nKey'
-    super(options)
-  }
-
-  extract(content) {
-    this.extractInterpolate(content)
-    this.extractTrans(content)
-    return this.keys
-  }
-
-  extractInterpolate(content) {
-    let matches
-    const regex = new RegExp(
-      '<Interpolate([^>]*\\s' + this.attr + '[^>]*)\\/?>',
-      'gi'
-    )
-
-    while (matches = regex.exec(content)) {
-      const attrs = this.parseAttributes(matches[1])
-      const key = attrs.keys
-      if (key) {
-        this.keys.push({ ...attrs.options, key })
+export const JSXParserExtension = Object.assign({}, walk.base, {
```

Comment:
This is fun. There's no built-ins for walking the JSX part of the tree. This defines what to do when you hit one of these new elements. We're taking it and walk.base and making a new object to use as the traps.